concordium-android-wallet-example/app/src/main/java/com/concordium/example/wallet/activities/IssueIdentityActivity.kt

```diff
@@ -68,10 +76,30 @@ class IssueIdentityActivity : ComponentActivity() {
         global: CryptographicParameters,
         storage: Storage
     ) {
-        val request = Requests.createIssuanceRequest(storage.getWallet(), provider, global)
-        val url = getIssuanceUrl(provider, request)
-        launchBrowserCustomTab(url)
-        storage.identityProviderIndex.set(provider.ipInfo.ipIdentity.toString())
+        CoroutineScope(Dispatchers.Default).launch {
+            try {
+                val request = Requests.createIssuanceRequest(storage.getWallet(), provider, global)
+                val url = getIssuanceUrl(provider, request)
+
+                val okHttpClientBuilder = OkHttpClient().newBuilder()
+                    .followRedirects(false)
+                    .followSslRedirects(false)
+                val client = okHttpClientBuilder.build()
+                val httpRequest = Request.Builder().url(url).build()
+
+                client.newCall(httpRequest).execute().use { response ->
+                    val redirectedUrl = response.header(HttpHeaders.LOCATION)
```

Comment:
I'm trying to do the same for CryptoX, and returning the header turned out to be not enough. `Location` may have either absolute or relative URL. The second case won't work. Can be tested with Notabene staging.

Reply:
Found out that in order to get the latest redirect URL, you can let the HTTP client follow redirects and query `Response.networkResponse`:
```kotlin
    fun getVerificationRedirectUrl(verificationStartUrl: String): String =
        httpClient
            .newCall(
                Request.Builder()
                    .url(verificationStartUrl)
                    .cacheControl(CacheControl.FORCE_NETWORK)
                    .build()
            )
            .execute()
            .networkResponse
            ?.request
            ?.url // This is the last requested URL after following all the redirects.
            ?.toString()
            ?: error("The identity provider did not redirect as expected")
```

Reply:
I will take a look at this tomorrow, but it was intentional to not follow redirects as there should be exactly one.

Reply:
My updated solution:
```kotlin
fun getVerificationRedirectUri(
    verificationStartUrl: String,
    redirectUriScheme: String,
): String =
    httpClient
        .newBuilder()
        .followRedirects(false)
        .followSslRedirects(false)
        .build()
        .newCall(
            Request.Builder()
                .url(verificationStartUrl)
                .cacheControl(CacheControl.FORCE_NETWORK)
                .build()
        )
        .execute()
        .let { response ->
            check(response.isRedirect) {
                "The identity provider did not redirect as expected"
            }

            val redirectLocation = response.header("Location")
                ?: error("Can't find the location in the identity provider response")

            return@let if (Uri.parse(redirectLocation).scheme in setOf(
                    "http",
                    "https",
                    redirectUriScheme,
                )
            ) {
                // Return the location if it is a URI with an allowed scheme:
                // HTTP or HTTPS to proceed with the verification on a web page,
                // or the app redirect URI scheme to get the instant result.
                redirectLocation
            } else {
                // Otherwise try to resolve the location against the base URL
                // in case it is a relative path.
                response.request.url.resolve(redirectLocation)?.toString()
                    ?: error("Can't resolve the redirect: $redirectLocation")
            }
        }
```

Reply:
By the way, found out that Digital Trust shoots `409 Conflict` instead of redirect in case of idCredPub duplication 🙃

Reply:
I will make a writeup of the issues with the identity providers, and then just pause this issue for now. If other wallets are going to implement this, then we need the protocol to be well-defined.